src/streaming/controllers/ProtectionController.js

```diff
@@ -418,8 +421,62 @@ MediaPlayer.dependencies.ProtectionController = function () {
                 data:event.data,
                 error:null
             });
+        },
+        
+        // Resolve all init data, for Adobe Access the init may need to be downloaded from
+        // a given URL in the mpd.
+        getPsshForSupportedKeySystems = function(supportedKS) {
+
+            var self = this,
+                supportedKSInit = {};
+            supportedKSInit[MediaPlayer.dependencies.protection.KeySystem.eventList.ENAME_PSSH_REQUEST_COMPLETE] = function(event) {
+                var ksIdx = 0,
+                    pending = false;
+                while( ksIdx < supportedKS.length) {
+                    if (event.data.requestData.schemeIdUri.toLowerCase() === supportedKS[ksIdx].ks.schemeIdURI) {
+                        if (!event.error) {
+                            supportedKS[ksIdx].initData = event.data.message;
+                        }
+                        else {
+                            self.log("DRM: Could not download PSSH data for keysystem : " + supportedKS[ksIdx].ks.schemeIdURI);
+                            supportedKS.splice(ksIdx);
+                            continue;
+                        }
+                        supportedKS[ksIdx].ks.unsubscribe(MediaPlayer.dependencies.protection.KeySystem.eventList.ENAME_PSSH_REQUEST_COMPLETE, supportedKSInit);
+                    }
+                    else if (supportedKS[ksIdx].initData === "pending") {
+                        pending = true;
+                    }
+                    
+                    ksIdx++;
+                }
+                
+                if (!pending) {
+                    pendingPsshData = false;
+
+                    selectKeySystem.call(self, supportedKS, true);
+                }
+                
+            };
+            
+            if (supportedKS && supportedKS.length > 0) {
+                var ksIdx = 0;
+                
+                for(ksIdx = 0; ksIdx < supportedKS.length; ++ksIdx) {
+                    if (supportedKS[ksIdx].initData === "pending") {
+                        supportedKS[ksIdx].ks.subscribe(MediaPlayer.dependencies.protection.KeySystem.eventList.ENAME_PSSH_REQUEST_COMPLETE, supportedKSInit);
+                        pendingPsshData = true;
+                    }
+                }
+            }
+            
+            if (!pendingPsshData) {
+                selectKeySystem.call(self, supportedKS, true);
+            }
+
         };
```

Comment:
@greg80303 I understand your concern about including/excluding DRM systems, which is why the actual code to download the init data is in the Access specific keySystem and this code takes care of pending init data in a generic way without notify and subscribe to the other keySystem files. 
W3C Promises can do this too with each KeySystem having a getInitData method but not sure about the architectural policy of mixing Promises with Dijon code where Promises are not mandated by any spec like EME.